src/jit/morph.cpp

```diff
@@ -8698,6 +8698,53 @@ GenTree* Compiler::fgMorphCall(GenTreeCall* call)
         }
     }
 
+    // Optimize Math.Pow(x, c)
+    if ((call->gtCallMoreFlags & GTF_CALL_M_SPECIAL_INTRINSIC) &&
+        info.compCompHnd->getIntrinsicID(call->gtCallMethHnd) == CORINFO_INTRINSIC_Pow)
+    {
+        noway_assert(call->fgArgInfo->ArgCount() == 2);
+        GenTree* arg0    = gtArgEntryByArgNum(call, 0)->node;
+        GenTree* arg1    = gtArgEntryByArgNum(call, 1)->node;
+        GenTree* newNode = nullptr;
+
+        if (arg1->IsCnsFltOrDbl())
+        {
+            noway_assert(arg0->TypeGet() == arg1->TypeGet());
+            GenTreeDblCon* powerCon = arg1->AsDblCon();
+
+            if (powerCon->gtDconVal == 2.0)
+            {
+                if (arg0->OperIs(GT_LCL_VAR))
+                {
+                    // Math.Pow(x, 2) -> x*x where x is a local variable
+                    newNode = gtNewOperNode(GT_MUL, powerCon->TypeGet(), arg0, gtCloneExpr(arg0));
+                }
+                else if (arg0->OperIs(GT_IND) && arg0->AsIndir()->Addr()->gtGetOp1()->OperIs(GT_LCL_VAR))
+                {
+                    // Math.Pow(x, 2) -> x*x where x is a field
```

Comment:
The way the code is written the only field that it will recognize is the first field of a struct (the field at offset 0). I doubt that was the intention.

Reply:
@mikedn heh, definitely not the intention, I wish I actually could try the "introduce a tmp local" scenario instead.